### PR TITLE
Support primitive type/classes in HubConnection.on()

### PR DESCRIPTION
**Pull request by wtgodbe**
Fixes https://github.com/dotnet/aspnetcore/issues/25088

Changes the `.on()` method so that it handles casting to primitives, similar to what we already do in `invoke()` and `stream()`. Also changes tests to test that this new behavior works.

### CODE REVIEW

File: src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java [hunk 1]
```diff
@@ -43,4 +43,18 @@ public static String appendQueryString(String original, String queryStringValue)
             throw new UnsupportedOperationException("Cannot handle type class: " + type.getClass());
         }
     }
+    
+    @SuppressWarnings("unchecked")
+	public static <T> T cast(Class<?> returnClass, Object obj) {
```
Comment (BrennanConroy) [inline src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java:48]:
```suggestion
    public static <T> T cast(Class<?> returnClass, Object obj) {
```

File: src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java [hunk 2]
```diff
@@ -43,4 +43,18 @@ public static String appendQueryString(String original, String queryStringValue)
...
+        // Primitive types can't be cast with the Class cast function
+        if (returnClass.isPrimitive()) {
+            return (T) obj;
+        } else {
+            return (T)returnClass.cast(obj);
+        }
+    }
+    
+	public static <T> T cast(Type returnType, Object obj) {
```
Comment (BrennanConroy) [inline src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java:57]:
```suggestion
    public static <T> T cast(Type returnType, Object obj) {
```

File: src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java [hunk 3]
```diff
@@ -43,4 +43,18 @@ public static String appendQueryString(String original, String queryStringValue)
...
+    	return cast(typeToClass(returnType), obj);
```
Comment (BrennanConroy) [inline src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java:58]:
```suggestion
        return cast(typeToClass(returnType), obj);
```

**Review by BrennanConroy — APPROVED** [review]
Nice!

File: src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java (hunk 1 above)
Comment (halter73) [reply src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java:48]: How did you notice this?

Comment (BrennanConroy) [reply src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java:48]: image

Comment (halter73) [reply src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java:48]: Interesting. They show up as the same width for me. Did you somehow configure a tab width of 6 on github?

**Review by halter73 — APPROVED** [review]

File: src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java (hunk 1 above)
Comment (BrennanConroy) [reply src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java:48]: Nope, and just looked with a different browser where I'm not logged in and it still shows a difference.

Comment (halter73) [reply src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java:48]: Ah. It's my Refined GitHub extension that makes the tabwidth 4 spaces.

Comment (wtgodbe) [reply src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/Utils.java]: I really need to configure Eclipse to treat tabs as spaces 😆

**Comment by wtgodbe** [comment]
@Pilchie can we take this for rc2? It fixes a bug where customers couldn't use primitive types in the `on` handlers of the java SignalR client (instead they had to use wrapper classes, e.g. `Integer.class` instead of `int.class`).

**Comment by Pilchie** [comment]
Approved for .NET 5 RC2.

**Pull request merged**